bebop_test/bebop_test.cc

```diff
@@ -72,7 +69,9 @@ int main()
         // display the drone's video stream on screen
         Video::Display display(drone.getVideoStream());
         do {
-            if (!joystick.update() && !display.update()) {
+            bool joyUpdated = joystick.update();
+            bool dispUpdated = display.update();
+            if (!joyUpdated && !dispUpdated) {
```

Comment:
Hah this is a good spot that I totally failed to make - nasty lazy-evaluation bug lurking there